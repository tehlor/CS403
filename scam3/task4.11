(define input-1 (make-wire))
(define input-2 (make-wire))
(define sum (make-wire))
(define carry (make-wire))
(probe 'sum sum)
(probe 'carry carry)
(half-adder input-1 input-2 sum carry)
(set-signal! input-1 0)
(propagate)
(set-signal! input-2 1)
(propagate)
(print "\nsum is " (get-signal sum))
(print "\ncarry is " (get-signal carry))
(println)
