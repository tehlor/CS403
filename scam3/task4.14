(define a (make-wire))
(define b (make-wire))
(define out (make-wire))
(probe 'out out)
(or-gate a b out)
(print "\n---------------")
(print "\n0 OR 0")
(print "\n---------------")
(set-signal! a 0)
(set-signal! b 0)
(propagate)
(print "\noutput is " (get-signal out))
(print "\n---------------")
(print "\n0 OR 1")
(print "\n---------------")
(set-signal! b 1)
(propagate)
(print "\noutput is " (get-signal out))
(print "\n---------------")
(print "\n1 OR 1")
(print "\n---------------")
(set-signal! a 1)
(propagate)
(print "\noutput is " (get-signal out))
(print "\n---------------")
(print "\n1 OR 0")
(print "\n---------------")
(set-signal! b 0)
(propagate)
(print "\noutput is " (get-signal out))
(println)
