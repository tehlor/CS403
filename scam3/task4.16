(define a (make-wire))
(define b (make-wire))
(define out (make-wire))
(probe 'out out)
(nor-gate a b out)
(print "\n---------------")
(print "\n0 NOR 0")
(print "\n---------------")
(set-signal! a 0)
(set-signal! b 0)
(propagate)
(print "\noutput is " (get-signal out))
(print "\n---------------")
(print "\n0 NOR 1")
(print "\n---------------")
(set-signal! b 1)
(propagate)
(print "\noutput is " (get-signal out))
(print "\n---------------")
(print "\n1 NOR 1")
(print "\n---------------")
(set-signal! a 1)
(propagate)
(print "\noutput is " (get-signal out))
(print "\n---------------")
(print "\n1 NOR 0")
(print "\n---------------")
(set-signal! b 0)
(propagate)
(print "\noutput is " (get-signal out))
(println)
