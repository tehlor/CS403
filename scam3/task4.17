(define a (make-wire))
(define b (make-wire))
(define out (make-wire))
(probe 'out out)
(xnor-gate a b out)
(print "\n---------------")
(print "\n0 XNOR 0")
(print "\n---------------")
(set-signal! a 0)
(set-signal! b 0)
(propagate)
(print "\noutput is " (get-signal out))
(print "\n---------------")
(print "\n0 XNOR 1")
(print "\n---------------")
(set-signal! b 1)
(propagate)
(print "\noutput is " (get-signal out))
(print "\n---------------")
(print "\n1 XNOR 1")
(print "\n---------------")
(set-signal! a 1)
(propagate)
(print "\noutput is " (get-signal out))
(print "\n---------------")
(print "\n1 XNOR 0")
(print "\n---------------")
(set-signal! b 0)
(propagate)
(print "\noutput is " (get-signal out))
(println)
