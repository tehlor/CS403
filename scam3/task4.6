(define a (make-wire))
(define b (make-wire))
(define out (make-wire))
(probe 'out out)
(xor-gate a b out)
(print "\n---------------")
(print "\n0 XOR 0")
(print "\n---------------")
(set-signal! a 0)
(set-signal! b 0)
(propagate)
(print "\noutput is " (get-signal out))
(print "\n---------------")
(print "\n0 XOR 1")
(print "\n---------------")
(set-signal! b 1)
(propagate)
(print "\noutput is " (get-signal out))
(print "\n---------------")
(print "\n1 XOR 1")
(print "\n---------------")
(set-signal! a 1)
(propagate)
(print "\noutput is " (get-signal out))
(print "\n---------------")
(print "\n1 XOR 0")
(print "\n---------------")
(set-signal! b 0)
(propagate)
(print "\noutput is " (get-signal out))
(println)
