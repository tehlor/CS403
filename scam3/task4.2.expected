
sum 0  New-value = 0
carry 0  New-value = 0
carry 6  New-value = 1
sum 6  New-value = 1
carry 12  New-value = 0
sum 12  New-value = 0
sum 18  New-value = 1
sum 24  New-value = 0
sum 48  New-value = 1
sum is 1
carry is 0
